pkg/controllers/vdb/dbremovesubcluster_reconciler.go

```diff
@@ -180,6 +203,24 @@ func (d *DBRemoveSubclusterReconciler) updateSubclusterStatus(ctx context.Contex
 	return vdbstatus.Update(ctx, d.VRec.Client, d.Vdb, refreshInPlace)
 }
 
+// updateAnnotationForReplicatedUpgrade updates the annotation ReplicatedUpgradeReplicaARemovedAnnotation
+// to indicate we have removed the subclusters in replica A successfully
+func (d *DBRemoveSubclusterReconciler) updateAnnotationForReplicatedUpgrade(ctx context.Context) error {
```

Comment:
This should be in `replicatedupgrade_reconciler just after `removesubcluster`.
Also, I suggest you use `vk8s.MetaUpdate` to update annotations.

Reply:
Why `vk8s.MetaUpdate` is better? Does it have a retry logic?